src/server/app.ts

```diff
@@ -67,6 +75,15 @@ class App {
             }
         });
     }
+
+    async close(): Promise<void> {
+        if (this.server) {
+            const httpTerminator = createHttpTerminator({ server: this.server });
+            await httpTerminator.terminate();
+
+            delete this.server;
```

Comment:
This is something we didn't have in other services. Why do need it and should we change the others accordingly?

Reply:
It is just a convenience operation, to not attempt to close the server if already done.

the existing codebase is improved in scim, ms-teams etc. to do such things in an elegant way. For example, `server` in zeplin-api is defined as `let server;` on file scope, we cannot use `delete` operator on such a variable, but we could set it to undefined. As I said, it is not mandatory but just a nice to have thing

Reply:
Also, in some of the other services, we did initialized the `httpTerminator` on server.init. Does it have a difference?

Reply:
according to the [source code](https://github.com/gajus/http-terminator/blob/master/src/factories/createInternalHttpTerminator.ts), it makes difference. it keeps the list of open sockets while server is listening.